### PR TITLE
Adding Happy Path test as PR check

### PR DESCRIPTION
**Pull request by rhopp**
Signed-off-by: Radim Hopp <rhopp@redhat.com>

### What does this PR do?

Frist tries to run Happy Path in this repo.

issue
https://github.com/eclipse/che/issues/13327

### CODE REVIEW

**Comment by benoitf** [comment]
did you tried to run happy path on github action runner ? or there is not enough memory ?

**Comment by rhopp** [comment]
> did you tried to run happy path on github action runner ? or there is not enough memory ?

Nope... didn't try that. Is there any specific reason, why you prefer github actions?

**Comment by benoitf** [comment]
>> did you tried to run happy path on github action runner ? or there is not enough memory ?

>@benoitf Nope... didn't try that. Is there any specific reason why you prefer github actions?

yes, it's faster (and it's easier to maintain and any committer can restart jobs)

**Comment by rhopp** [comment]
[ci-test-happy-path]

File: .ci/cico-happy-path-pr-test.sh [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+obtainUserToken
+
+createTestWorkspaceAndRunTest
+
+getOpenshiftLogs
+
+archiveArtifacts "che-plugin-registry-prcheck"
+
+if [[ "$IS_TEST_FAILED" == "true" ]]; then 
+  FAIL_MESSAGE="Faile on running tests. Please check logs or contact QE team."
```
Comment (dmytro-ndp) [inline .ci/cico-happy-path-pr-test.sh:121]: Typo: "Faile".
Do we have Eclipse Che QE mailing list to contact us?

File: .ci/cico-happy-path-pr-test.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# Start Che
+
+if chectl server:start --listr-renderer=verbose -a operator -p openshift --k8spodreadytimeout=360000 --che-operator-cr-patch-yaml=/tmp/che-cr-patch.yaml; then
+    echo "Started succesfully"
+    oc get checluster -o yaml
+else
+    getOpenshiftLogs
+    archiveArtifacts "che-plugin-registry-prcheck"
+    FAIL_MESSAGE="Failed while starting chectl. Please check logs."
+    exit 1337
```
Comment (dmytro-ndp) [inline .ci/cico-happy-path-pr-test.sh:107]: If there is a special reason to return 1337 instead of just 1?

File: .ci/cico-happy-path-pr-test.sh [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+load_jenkins_vars
+install_deps
+
+# Build & push.
+
+export TAG="PR-${ghprbPullId}"
+export IMAGE_NAME="quay.io/eclipse/che-plugin-registry:$TAG"
+build_and_push
+
+export FAIL_MESSAGE="Build passed. Image is available on $IMAGE_NAME"
```
Comment (dmytro-ndp) [inline .ci/cico-happy-path-pr-test.sh:74]: IMHO it doesn't look like a fail message but rather as a successful build message.
Doesn't "echo" work here instead?

File: .ci/cico_common.sh [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  chmod 600 ./artifacts.key
+  chown $(whoami) ./artifacts.key
+  mkdir -p ./che/${JOB_NAME}/${BUILD_NUMBER}
+  cp -R ./report ./che/${JOB_NAME}/${BUILD_NUMBER}/ | true
+  rsync --password-file=./artifacts.key -Hva --partial --relative ./che/${JOB_NAME}/${BUILD_NUMBER} devtools@artifacts.ci.centos.org::devtools/
+}
+
+
+function obtainUserToken() {
+
```
Comment (dmytro-ndp) [inline .ci/cico_common.sh:123]: redundant empty line

File: .ci/cico-happy-path-pr-test.sh [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+function getOpenshiftLogs() {
+  mkdir -p /root/payload/report/oc-logs
+  cd /root/payload/report/oc-logs
+  for POD in $(oc get pods -o name); do
+    for CONTAINER in $(oc get ${POD} -o jsonpath="{.spec.containers[*].name}"); do
+      oc logs ${POD} -c ${CONTAINER} |tee $(echo ${POD}-${CONTAINER}.log | sed 's|pod/||g')
+    done
+  done
+  echo "======== oc get events ========"
+  oc get events |tee get_events.log
```
Comment (dmytro-ndp) [inline .ci/cico-happy-path-pr-test.sh:22]: `|tee get_events.log` > `| tee get_events.log`

File: .ci/cico-happy-path-pr-test.sh [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  echo "======== oc get all ========"
+  oc get all |tee get_all.log
```
Comment (dmytro-ndp) [inline .ci/cico-happy-path-pr-test.sh:24]: `|tee get_all.log` > `| tee get_all.log` is more readable

**Comment by dmytro-ndp** [comment]
It would be nice to see links to test logs in `Happy path tests failed.` comment on PR page.

**Comment by dmytro-ndp** [comment]
What about archiving failed happy path tests report which being stored into `tests/e2e/report/**` directory?

File: .ci/cico-happy-path-pr-test.sh (hunk 2 above)
Comment (rhopp) [reply .ci/cico-happy-path-pr-test.sh:105]: No specific reason... just a joke without any effect on functionality ;-)
(https://en.wikipedia.org/wiki/Leet)

**Comment by rhopp** [comment]
This should be ready for review.
May be interesting for you, @JPinkney  as I see you have draft PR doing something simmilar.

**Comment by JPinkney** [comment]
@rhopp my changes for tests in the plugin registry are going to be picked up by @ericwill's team this sprint. I think if these get merged beforehand we can re-use a lot of this code as well

**Comment by ericwill** [comment]
cc @svor

File: .ci/cico-happy-path-pr-test.sh [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  echo "======== oc get all ========"
+  oc get all | tee get_all.log
+  oc get checluster -o yaml | tee get_checluster.log || true
+}
+
+createTestWorkspaceAndRunTest() {
+  CHE_URL=$(oc get checluster eclipse-che -o jsonpath='{.status.cheURL}')
+  ### Create workspace
+  echo "====== Create test workspace ======"
+  chectl workspace:create --start --access-token "$USER_ACCESS_TOKEN" --devfile=https://raw.githubusercontent.com/eclipse/che/master/tests/e2e/files/happy-path/happy-path-workspace.yaml
```
Comment (flacatus) [inline .ci/cico-happy-path-pr-test.sh:32]: Typo:
-  If you put the access token in ``CHE_ACCESS_TOKEN`` environment  you don't need to use anymore --access-token flag.

**Pull request merged**